### PR TITLE
feat: add apps api hooks

### PR DESCRIPTION
**Pull request by jkdowdle**
## Changes

  - Your changes...

### CODE REVIEW

**Review by sternetj — APPROVED** [review]

File: src/types/apps-rest-types.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/**
```
Comment (aecorredor) [inline src/types/apps-rest-types.ts:1]: This file should be named `apps-auth-rest-types.ts`, right?

File: src/types/apps-rest-types.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * While performing authorization through an external user-agent (browser)
+ * is recommended and the only way to be OAuth 2.0 compliant, there are
+ * publicly available endpoints that can be used to build a fully native
+ * experience.
+ */
+export type AppsAPIEndpoints = {
```
Comment (aecorredor) [inline src/types/apps-rest-types.ts:7]: We should name this `AppsAuthAPIEndpoints`, right?

**Review by aecorredor — COMMENTED** [review]
Don't we want `AppsAuth` instead of just `Apps` in all occurrences in this PR?

File: src/types/apps-rest-types.ts (hunk 1 above)
Comment (jkdowdle) [reply src/types/apps-rest-types.ts:1]: @aecorredor I think Apps is more appropriate if we ever wanted to add anything else under this origin `https://apps.us.lifeomic.com`, not just the `/auth` routes

**Pull request merged**